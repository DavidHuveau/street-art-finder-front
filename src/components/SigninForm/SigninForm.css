.Signin {
  border: solid 1px#ddd;
  border-radius: 4px;
  margin-bottom: 20px;
}

.Signin .card-header {
  text-align: center;
  padding: 10px 15px;
}

.Signin .card-body {
  padding-left: 16px;
}

.Signin .form-row {
  margin-right: 0px !important;
  margin-left: 0px !important;
}

.Signin .form-label {
  font-weight: 700;
}