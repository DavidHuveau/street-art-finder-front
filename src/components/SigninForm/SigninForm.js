import React, { Component } from "react";
import "./SigninForm.css";
import { Form, Button, Card, Container } from "react-bootstrap";

class SignInForm extends Component {
  constructor() {
    super();

    ["onChangeValue", "onSubmit", "renderForm"].forEach(
      fn => (this[fn] = this[fn].bind(this))
    );

    // this.state = {
    //   login: "toto",
    //   password: "1234"
    // };
    this.state = {
      login: "",
      password: ""
    };
  }

  onChangeValue(e) {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  }

  onSubmit(e) {
    e.preventDefault();

    const { login, password } = this.state;
    const { onSubmit } = this.props;

    onSubmit({ login: login, password: password });
  }

  renderForm() {
    const { login, password } = this.state;

    return (
      <Form
      methode="POST"
      onSubmit={this.onSubmit}
      >
        <Form.Group controlId="formGroupLogin">
          <Form.Label>Login</Form.Label>
          <Form.Text className="text-muted">Your login.</Form.Text>
          <Form.Control
            type="text"
            placeholder="Enter a login..."
            value={login}
            onChange={this.onChangeValue}
            name="login"
            required
          />
        </Form.Group>
        <Form.Group controlId="formGroupPassword">
          <Form.Label>Password</Form.Label>
          <Form.Text className="text-muted">Your password.</Form.Text>
          <Form.Control
            type="password"
            placeholder="Enter a password..."
            value={password}
            onChange={this.onChangeValue}
            name="password"
            required
          />
        </Form.Group>
      </Form>
    );
  }

  renderSubmitButton() {
    return (
      <div className="float-right">
        <Button type="submit" color="primary">
          Login
        </Button>
      </div>
    );
  }

  render() {
    return (
      <Container style={{ marginBottom: "70px"}}>
        <Card bsPrefix="Signin" className="down-lg">
          <Card.Header>
            <strong>Sign In</strong>
          </Card.Header>
          <Card.Body>
            {this.renderForm()}
          </Card.Body>
        </Card>
        {this.renderSubmitButton()}
      </Container>
    );
  }
}

export default SignInForm;
