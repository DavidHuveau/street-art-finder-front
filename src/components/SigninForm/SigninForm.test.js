import React from "react";
import renderer from "react-test-renderer";
import SigninForm from "./SigninForm";

describe("Check Components", () => {
  it("Component SigninForm", () => {
    const tree = renderer
      .create(<SigninForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})