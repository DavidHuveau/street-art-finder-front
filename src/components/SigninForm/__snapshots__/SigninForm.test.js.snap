// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Components Component SigninForm 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="container"
    style={
      Object {
        "marginBottom": "70px",
      }
    }
  >
    <div
      className="down-lg Signin"
    >
      <div
        className="card-header"
      >
        <strong>
          Sign In
        </strong>
      </div>
      <div
        className="card-body"
      >
        <div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formGroupLogin"
            >
              Login
            </label>
            <small
              className="text-muted form-text"
            >
              Your login.
            </small>
            <input
              className="form-control"
              id="formGroupLogin"
              name="login"
              onChange={[Function]}
              placeholder="Enter a login..."
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formGroupPassword"
            >
              Password
            </label>
            <small
              className="text-muted form-text"
            >
              Your password.
            </small>
            <input
              className="form-control"
              id="formGroupPassword"
              name="password"
              onChange={[Function]}
              placeholder="Enter a password..."
              required={true}
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="float-right"
    >
      <button
        className="btn btn-primary"
        color="primary"
        disabled={false}
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</form>
`;
